### PR TITLE
Fix gpu work cts gpu_work case issue

### PR DESCRIPTION
**Pull request by beckwen**
For quick fix this issue, implement one workaround

Tracked-On: OAM-125102

### CODE REVIEW

File: drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c [hunk 1]
```diff
@@ -3368,11 +3371,18 @@ i915_gem_do_execbuffer(struct drm_device *dev,
 	struct sync_file *out_fence = NULL;
 	int out_fence_fd = -1;
 	int err;
+    u64 start_time = ktime_get_raw_ns();
```
Comment (feijiang1) [inline drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c:3374]: tab issue

File: drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c [hunk 2]
```diff
@@ -3368,11 +3371,18 @@ i915_gem_do_execbuffer(struct drm_device *dev,
...
 
 	BUILD_BUG_ON(__EXEC_INTERNAL_FLAGS & ~__I915_EXEC_ILLEGAL_FLAGS);
 	BUILD_BUG_ON(__EXEC_OBJECT_INTERNAL_FLAGS &
 		     ~__EXEC_OBJECT_UNKNOWN_FLAGS);
 
+    const struct cred* cred = get_current_cred();
+    const unsigned int uid = cred->euid.val;
+    put_cred(cred);
+    trace_gpu_work_period(i915->drm.primary->index, uid,
```
Comment (feijiang1) [inline drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c:3383]: hard code the gpu time, need add more description for this workaround

**Review by feijiang1 — COMMENTED** [review]
The commit also need add more description, what's the problem in cts test, what's the workaround......

File: drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c (hunk 1 above)
Comment (beckwen) [reply drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c]: Done

File: drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c (hunk 2 above)
Comment (beckwen) [reply drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c]: Done

File: drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c [hunk 3]
```diff
@@ -3368,11 +3371,19 @@ i915_gem_do_execbuffer(struct drm_device *dev,
 	struct sync_file *out_fence = NULL;
 	int out_fence_fd = -1;
 	int err;
+	u64 start_time = ktime_get_raw_ns();
 
 	BUILD_BUG_ON(__EXEC_INTERNAL_FLAGS & ~__I915_EXEC_ILLEGAL_FLAGS);
 	BUILD_BUG_ON(__EXEC_OBJECT_INTERNAL_FLAGS &
 		     ~__EXEC_OBJECT_UNKNOWN_FLAGS);
 
+	/* Add one Workaround to pass the cts module GpuMetrics case com.android.cts.graphics.GpuWorkDumpsysTest*/
```
Comment (feijiang1) [inline drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c:3380]: the line should be too long. Kernel has strict coding style requirements, it is better to check the patch diff with kernel/linux-intel-lts2021/scripts/checkpatch.pl

Comment (feijiang1) [reply drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c:3380]: also need add more workaround description, for example, you hard code the execution time as 100ms etc

**Review by feijiang1 — APPROVED** [review]
LGTM

**Review by xyzhao2018 — COMMENTED** [review]
You PR also included some patches of virtio-camera, please re-sync your branch and submit again.

**Pull request closed without merging**